src/frontend/mame/ui/dirmenu.cpp

```diff
@@ -406,12 +395,12 @@ void menu_add_change_folder::handle()
 
 				// from current item to the end
 				for (entry = cur_selected; entry < item.size(); entry++)
-					if (item[entry].ref != nullptr && m_search[0] != 0)
+					if (item[entry].ref != nullptr && !m_search.empty())
 					{
 						int match = 0;
-						for (int i = 0; i < ARRAY_LENGTH(m_search); i++)
+						for (int i = 0; i < m_search.size(); i++)
```

Comment:
You've subtly changed behaviour of this loop.  Previously it would compare the terminator as well, meaning if you searched for "hi", it would get a score of 2 for "hit" and 3 for "hi".  You've broken that.